### PR TITLE
add(force flag to new demux pp flow)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR adds a force flag to the demux post processing flow that in case the flow cell is already been marked for delivery it will re-run the process.
part of:
https://github.com/Clinical-Genomics/project-planning/issues/482

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by ChrOertlin** [comment]
# no force flag:
image

# force flag flow starts even though marked for delivery:
image

File: tests/meta/demultiplex/test_validation.py [hunk 1]
```diff
@@ -209,19 +208,32 @@ def test_validate_demultiplexing_complete_no_error(tmp_path):
...
-        validate_flow_cell_delivery_status(tmp_path)
+        validate_flow_cell_delivery_status(flow_cell_output_directory=tmp_path, force=False)
+
+
+def test_validate_flow_cell_delivery_status_forced(tmp_path: Path):
+    # GIVEN a path where DELIVERY file is present
+    (tmp_path / DemultiplexingDirsAndFiles.DELIVERY).touch()
+
+    # WHEN validating the flow cell delivery status
+    # THEN it should raise a FlowCellError
```
Comment (seallard) [inline tests/meta/demultiplex/test_validation.py:236]: Update doc string

**Review by seallard — APPROVED** [review]
Looks good! 

This change means that the new flow will run when the force flag is provided. It seems like it could cause new unexpected side effects, but I can't think of any. Maybe there will be an error when we try to create the `delivery.txt` file and it already exists?

File: tests/meta/demultiplex/test_validation.py (hunk 1 above)
Comment (ChrOertlin) [inline tests/meta/demultiplex/test_validation.py:236]:
```suggestion
    # THEN it should not raise a FlowCellError
```

**Comment by ChrOertlin** [comment]
> Looks good!
> 
> This change means that the new flow will run when the force flag is provided. It seems like it could cause new unexpected side effects, but I can't think of any. Maybe there will be an error when we try to create the `delivery.txt` file and it already exists?

The test did not raise any error even though the file existed. I think it simply overwrites if there are no checks

**Review by Vince-janv — APPROVED** [review]
Cool! Just one concern. I think `touch()` will crash when it tries to create the delivery.txt file again and a file is already there. Did this happen in the test?

**Comment by ChrOertlin** [comment]
I just re-ran the test it does not complain about the existing file and just overwrites the file

**Comment by ChrOertlin** [comment]
stage:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7
done.
Log deploy... done.
cg, version 40.4.1

Prod:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7
done.
Log deploy... done.
cg, version 40.4.1

**Pull request merged**